spec/help_command_handler_spec.rb

```diff
@@ -0,0 +1,15 @@
+require './lib/help_command_handler'
+
+RSpec.describe HelpCommandHandler do
+    subject(:help_command_handler) { HelpCommandHandler.new }
+
+    describe "#print_help_message" do
+        context 'when initialising' do
+            let(:help_message) { 'Try one of these commands: <show>, <search>' }
+
+            it 'shows a list of available commands' do
+                expect{help_command_handler.print_help_message}.to output(help_message).to_stdout
```

Comment:
Love the fact that I can look at the tests and figure out how the code works, this is how test should be used.

Because you class has a dependency on stdout it makes it tricky to test, the way you tested it is one way to do it.

The way I think about it as that HelpCommandHandler is part of your business logic and you should keep it as decoupled as possible for the presentation layer. In this case the presentation is stdout.

One other way to think about it is that the command handler result should be usable from another module int this case beacause it outputs to stdout one can't use this handle as a building block for a more complex business logic.

My suggestion is to have the handler return a result and then use that result from some other place for presentation.